### PR TITLE
Remove client side navigation from Products beta block

### PR DESCRIPTION
**Pull request by imanish003**
Changes:
- Removed the `add_navigation_id_directive` method and its associated filter. This method previously added a `data-wc-navigation-id` attribute to the query block for client-side navigation, which is no longer required.
- Removed the `add_navigation_link_directives` method and its associated filter. This method previously added interactive directives to pagination links inside the Query Pagination block. This specific functionality has been removed.
- Introduced a new method `mark_block_as_interactive`, which is designed to mark the Product Query as an interactive region so that interactive elements can work inside it. This is achieved using the 'data-wc-interactive' attribute.

#### Other Checks

- [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
- [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

### Testing

1. Create a new post
2. Add Products (Beta) block & publish it
3. On Frontend, verify that
    1. Client-side navigation isn't there anymore i.e. changing the page should also refresh the page
    2. Verify that Product button(Add to cart) still works as expected.

Here is a quick demo:

https://github.com/woocommerce/woocommerce-blocks/assets/16707866/884b3e32-cabd-425e-b993-ca985dc3308b

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

### WooCommerce Visibility

* [X] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Remove client side navigation from Products beta block

### CODE REVIEW

**Review by gigitux — APPROVED** [review]
LGTM

**Comment by luisherranz** [comment]
> Introduced a new method mark_block_as_interactive, which is designed to mark the Product Query as an interactive region so that interactive elements can work inside it. This is achieved using the 'data-wc-interactive' attribute.

Instead of doing this, if the Products (beta) block doesn't have any interactivity in itself, you should add `data-wc-interactive` to the Product Button blocks instead. That way, the Product Button blocks will also work when they are used outside of the Products (beta) block (like in the product template, if that is a possibility).

**Comment by imanish003** [comment]
> > Introduced a new method mark_block_as_interactive, which is designed to mark the Product Query as an interactive region so that interactive elements can work inside it. This is achieved using the 'data-wc-interactive' attribute.
> 
> Instead of doing this, if the Products (beta) block doesn't have any interactivity in itself, you should add `data-wc-interactive` to the Product Button blocks instead. That way, the Product Button blocks will also work when they are used outside of the Products (beta) block (like in the product template, if that is a possibility).

I think it’s much better approach to have `data-wc-interactive` on all interactive elements, in our case Product Button. Let me try doing it. Thanks 🙌🏻

File: src/BlockTypes/ProductButton.php [hunk 1]
```diff
@@ -192,7 +192,7 @@ protected function render( $attributes, $content, $block ) {
 			return apply_filters(
 				'woocommerce_loop_add_to_cart_link',
 				strtr(
-					'<div class="wp-block-button wc-block-components-product-button {classes} {custom_classes}"
+					'<div data-wc-interactive=true class="wp-block-button wc-block-components-product-button {classes} {custom_classes}"
```
Comment (luisherranz) [inline src/BlockTypes/ProductButton.php]: It's a boolean attribute, you don't need it to be true.

```suggestion
					'<div data-wc-interactive class="wp-block-button wc-block-components-product-button {classes} {custom_classes}"
```

**Review by luisherranz — APPROVED** [review]
Tested and it works fine 🙂👍️

File: src/BlockTypes/ProductButton.php (hunk 1 above)
Comment (imanish003) [reply src/BlockTypes/ProductButton.php]: Good catch! Made the changes in 088604d, thanks 🙌🏻

**Pull request merged**